app/src/App.tsx

```diff
@@ -458,27 +456,36 @@ export default function App() {
         }
       } else if (event.key == "c") {
         if (selectedPoint !== null) {
-          setOrbitControlsTarget(selectedPoint);
+          setCameraAndControls((prev: any) => ({
+            ...prev,
+            target: selectedPoint,
+          }));
         } else {
           if (currentFrame.positions.length > 0) {
             const center = getCentroid(currentFrame.positions, selectedIds);
-            setOrbitControlsTarget(center);
+            console.log(center);
+            setCameraAndControls((prev: any) => ({
+              ...prev,
+              target: center,
+            }));
           }
         }
       } else if (event.key == "o") {
         const origin = {
           position: [10, 10, 10],
           target: getCentroid(currentFrame.positions, new Set()),
         };
-        setOrbitControlsTarget(new THREE.Vector3(...origin.target));
-        setCameraPosition(new THREE.Vector3(...origin.position));
-        if (controlsRef.current && cameraRef.current) {
-          controlsRef.current.enabled = false;
-          cameraRef.current.position.set(...origin.position);
-          setCameraRoll(null);
-          controlsRef.current.update();
-          controlsRef.current.enabled = true;
-        }
+        setCameraAndControls({
+          camera: new THREE.Vector3(...origin.position),
+          target: new THREE.Vector3(...origin.target),
+        });
+        // if (controlsRef.current && cameraRef.current) {
+        //   controlsRef.current.enabled = false;
+        //   cameraRef.current.position.set(...origin.position);
+        //   setCameraRoll(null);
+        //   controlsRef.current.update();
+        //   controlsRef.current.enabled = true;
+        // }
```

Comment:
move the `o` and `c` to the camera module, also adapt the `o` to make use of the FOV